DfE.FindInformationAcademiesTrusts/Services/Export/ExportService.cs

```diff
@@ -96,46 +101,202 @@ private static byte[] GenerateSpreadsheet(
+            var academyDetails = academies[i];
+            var urn = academyDetails.Urn;
+            var ofstedData = academiesOfstedRatings.SingleOrDefault(x => x.Urn == urn);
+
+            var previousRating = ofstedData?.PreviousOfstedRating ?? OfstedRating.NotInspected;
+            var currentRating = ofstedData?.CurrentOfstedRating ?? OfstedRating.NotInspected;
+
+            // School Name
+            SetTextCell(worksheet, i + 4, 1, academyDetails.EstablishmentName ?? string.Empty);
+
+            // Date Joined Trust
+            SetDateCell(worksheet, i + 4, 2, ofstedData?.DateAcademyJoinedTrust);
+
+            // Current Inspection Date
+            SetDateCell(worksheet, i + 4, 3, currentRating.InspectionDate);
+
+            // Before/After Joining
+            SetTextCell(worksheet, i + 4, 4,
+                IsOfstedRatingBeforeOrAfterJoining(currentRating.OverallEffectiveness,
+                    ofstedData?.DateAcademyJoinedTrust, currentRating.InspectionDate));
+
+            // Current Ratings
+            SetTextCell(worksheet, i + 4, 5, currentRating.QualityOfEducation.ToDisplayString());
+            SetTextCell(worksheet, i + 4, 6, currentRating.BehaviourAndAttitudes.ToDisplayString());
+            SetTextCell(worksheet, i + 4, 7, currentRating.PersonalDevelopment.ToDisplayString());
+            SetTextCell(worksheet, i + 4, 8, currentRating.EffectivenessOfLeadershipAndManagement.ToDisplayString());
+            SetTextCell(worksheet, i + 4, 9, currentRating.EarlyYearsProvision.ToDisplayString());
+            SetTextCell(worksheet, i + 4, 10, currentRating.SixthFormProvision.ToDisplayString());
+
+            // Previous Inspection Date
+            SetDateCell(worksheet, i + 4, 11, previousRating.InspectionDate);
+
+            // Previous Ratings
+            SetTextCell(worksheet, i + 4, 12, previousRating.QualityOfEducation.ToDisplayString());
+            SetTextCell(worksheet, i + 4, 13, previousRating.BehaviourAndAttitudes.ToDisplayString());
+            SetTextCell(worksheet, i + 4, 14, previousRating.PersonalDevelopment.ToDisplayString());
+            SetTextCell(worksheet, i + 4, 15, previousRating.EffectivenessOfLeadershipAndManagement.ToDisplayString());
+            SetTextCell(worksheet, i + 4, 16, previousRating.EarlyYearsProvision.ToDisplayString());
+            SetTextCell(worksheet, i + 4, 17, previousRating.SixthFormProvision.ToDisplayString());
+
+            // Before/After Joining 
+            SetTextCell(worksheet, i + 4, 18,
+                IsOfstedRatingBeforeOrAfterJoining(previousRating.OverallEffectiveness,
+                    ofstedData?.DateAcademyJoinedTrust, previousRating.InspectionDate));
+
+            // Safeguarding Effective
+            SetTextCell(worksheet, i + 4, 19, currentRating.SafeguardingIsEffective.ToDisplayString());
+
+            // Category of Concern
+            SetTextCell(worksheet, i + 4, 20, currentRating.CategoryOfConcern.ToDisplayString());
```

Comment:
Just a suggestion, I think this could be separated out into a separate function, something like `GenerateOfstedRow`. This current function is already quite long and could do with splitting up

Reply:
Done